File: leetcode/problems/python3/0045. Jump Game II/solve.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        L = len(nums)
        if L == 1:
            return 0
        steps = 1
        farthest = 0
        check_end = 0
        for i, n in enumerate(nums):
            farthest = max(i + n, farthest)
            if farthest >= L - 1:
                break
            if i == check_end:
                steps += 1
                check_end = farthest
        return steps

File: leetcode/problems/java/0045. Jump Game II/solve.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        if (n <= 1) {
            return 0;
        }

        int steps = 1;
        int farthest = 0;
        int checkEnd = 0;
        for (int i = 0; i < n - 1; i++) {
            farthest = Math.max(i + nums[i], farthest);
            if (farthest >= n - 1) {
                break;
            }
            if (i == checkEnd) {
                steps++;
                checkEnd = farthest;
            }
        }
        return steps;
    }
}